test/sagemaker_tests/pytorch/training/conftest.py

```diff
@@ -564,6 +564,26 @@ def skip_p5_tests(instance_type, processor, ecr_image):
             pytest.skip("P5 EC2 instance require CUDA 12.0 or higher.")
 
 
+@pytest.fixture(autouse=True)
+def skip_smdataparallel_p5_tests(instance_type, processor, ecr_image):
+    """SMDDP tests are broken for PyTorch 2.1 on p5 instances, so we should skip"""
+    smdataparallel_skip = False
+    skip_dict = {"==2.1.*": ["cu121"]}
+    if _validate_pytorch_framework_version(
```

Comment:
you can change this to
```
    if _validate_pytorch_framework_version(
        request, processor, ecr_image, "skip_smddataparallel_test", skip_dict
    ) and "p5." in instance_type:
        pytest.skip"<INSERT REASON"
```

The second if statement is unnecessary if we're already using `_validate_pytorch_framework_version()` functions.